integration-tests/testsetups/keeper_benchmark.go

```diff
@@ -460,37 +478,47 @@ func (k *KeeperBenchmarkTest) DeployBenchmarkKeeperContracts(
 ) {
 	l := utils.GetTestLogger(t)
 	registryVersion := k.Inputs.RegistryVersions[index]
+	k.Inputs.KeeperRegistrySettings.RegistryVersion = registryVersion
 	upkeep := k.Inputs.Upkeeps
-	registry := actions.DeployKeeperRegistry(t, k.contractDeployer, k.chainClient,
-		&contracts.KeeperRegistryOpts{
-			RegistryVersion: registryVersion,
-			LinkAddr:        k.linkToken.Address(),
-			ETHFeedAddr:     k.ethFeed.Address(),
-			GasFeedAddr:     k.gasFeed.Address(),
-			TranscoderAddr:  actions.ZeroAddress.Hex(),
-			RegistrarAddr:   actions.ZeroAddress.Hex(),
-			Settings:        *k.Inputs.KeeperRegistrySettings,
-		},
+	var (
+		registry  contracts.KeeperRegistry
+		registrar contracts.KeeperRegistrar
 	)
 
-	// Fund the registry with 1 LINK * amount of AutomationConsumerBenchmark contracts
-	err := k.linkToken.Transfer(registry.Address(), big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(int64(k.Inputs.Upkeeps.NumberOfUpkeeps))))
-	require.NoError(t, err, "Funding keeper registry contract shouldn't fail")
+	if registryVersion <= 3 {
```

Comment:
Nit: I would recommend adding a comment to state which versions are <= 3 or even use the specific versions in code to make code more readable.